Sorbet/main.rb

```diff
@@ -0,0 +1,326 @@
+def predicate_2way_success_g(x)
+  if predicate_2way_success_f(x)
+    x.length
+  else
+    x + 0
+  end
+end
+
+sig { params(x: T.any(String, Integer)).returns(T::Boolean) }
+def predicate_2way_failure_f(x)
+  x.is_a?(String)
+end
+
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def predicate_2way_failure_g(x)
+  if predicate_2way_failure_f(x)
+    x.is_nan # Expected error: is_nan not defined for String
+  else
+    x + 0
+  end
+end
+
+# Example predicate_1way
+sig { params(x: T.any(String, Integer)).returns(T::Boolean) }
+def predicate_1way_success_f(x)
+  x.is_a?(Integer) && x > 0
+end
+
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def predicate_1way_success_g(x)
+  if predicate_1way_success_f(x)
+    x + 1
+  else
+    0
+  end
+end
+
+sig { params(x: T.any(String, Integer)).returns(T::Boolean) }
+def predicate_1way_failure_f(x)
+  x.is_a?(Integer) && x > 0
+end
+
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def predicate_1way_failure_g(x)
+  if predicate_1way_failure_f(x)
+    x + 1
+  else
+    x.length # Expected error: length not defined for String | Integer
+  end
+end
+
+# Example predicate_checked
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(T::Boolean) }
+def predicate_checked_success_f(x)
+  x.is_a?(String)
+end
+
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(T::Boolean) }
+def predicate_checked_success_g(x)
+  !predicate_checked_success_f(x)
+end
+
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(T::Boolean) }
+def predicate_checked_failure_f(x)
+  x.is_a?(String) || x.is_a?(Integer)
+end
+
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(T::Boolean) }
+def predicate_checked_failure_g(x)
+  x.is_a?(Integer)
+end
+
+# Example object_properties
+sig { params(x: { a: T.untyped }).returns(Integer) }
+def object_properties_success_f(x)
+  if x[:a].is_a?(Integer)
+    x[:a]
+  else
+    0
+  end
+end
+
+sig { params(x: { a: T.untyped }).returns(Integer) }
+def object_properties_failure_f(x)
+  if x[:a].is_a?(String)
+    x[:a] # Expected error: String not assignable to Integer
+  else
+    0
+  end
+end
+
+# Example tuple_elements
+sig { params(x: [T.untyped, T.untyped]).returns(Integer) }
+def tuple_elements_success_f(x)
+  if x[0].is_a?(Integer)
+    x[0]
+  else
+    0
+  end
+end
+
+sig { params(x: [T.untyped, T.untyped]).returns(Integer) }
+def tuple_elements_failure_f(x)
+  if x[0].is_a?(Integer)
+    x[1].length # Expected error: length not defined for T.untyped
+  else
+    0
+  end
+end
+
+# Example tuple_length
+sig { params(x: T.any([Integer, Integer], [String, String, String])).returns(Integer) }
+def tuple_length_success_f(x)
+  if x.length == 2
+    x[0] + x[1]
+  else
+    x[0].length
+  end
+end
+
+sig { params(x: T.any([Integer, Integer], [String, String, String])).returns(Integer) }
+def tuple_length_failure_f(x)
+  if x.length == 2
+    x[0] + x[1]
```

Comment:
no failure ... I guess data structure elements always have the dynamic type in Sorbet.